PowerEditor/src/MISC/Common/Common.cpp

```diff
@@ -747,4 +747,97 @@ generic_string stringJoin(const std::vector<generic_string> &strings, generic_st
 		}
 	}
 	return joined;
+}
+
+int stoiStrict(const generic_string &input)
+{
+	if (input.empty())
+	{
+		throw std::invalid_argument("Empty input.");
+	}
+	else
+	{
+		// Check minus characters.
+		const int minuses = std::count(input.begin(), input.end(), TEXT('-'));
+		if (minuses > 1)
+		{
+			throw std::invalid_argument("More than one minus sign.");
+		}
+		else if (minuses == 1 && input[0] != TEXT('-'))
+		{
+			throw std::invalid_argument("Minus sign must be first.");
+		}
+
+		// Check for other characters which are not allowed.
+		if (input.find_first_not_of(TEXT("-0123456789")) != std::string::npos)
+		{
+			throw std::invalid_argument("Invalid character found.");
+		}
+
+		return std::stoi(input);
+	}
+}
+
+bool allLinesAreNumeric(const std::vector<generic_string> &lines)
+{
+	const auto endit = lines.end();
+	for (auto it = lines.begin(); it != endit; ++it)

```

Comment:
Why the C++03 style iterator loop? There's nothing wrong with it, but I imagine a C++11 style ranged for would be cleaner here.

à la
```
for (const auto& line : lines)
{
    try
    {
        stoiStrict(line);
    }
    catch (std::invalid_argument&)
    {
        return false;
    }
    catch (std::out_of_range&)
    {
        return false;
    }
}
return true;
```

...totally nitpicking, of course.

(also)Might as well use `lines.cbegin()`/`lines.cend()`, if the ranged for doesn't seem better to you.